pkg/registry/generic/registry/storage_factory.go

```diff
@@ -31,7 +31,24 @@ func StorageWithCacher(
 	resourcePrefix string,
 	scopeStrategy rest.NamespaceScopedStrategy,
 	newListFunc func() runtime.Object) storage.Interface {
-	return storage.NewCacher(
-		storageInterface, capacity, etcdstorage.APIObjectVersioner{},
-		objectType, resourcePrefix, scopeStrategy, newListFunc)
+
+	config := storage.CacherConfig{
+		CacheCapacity:  capacity,
+		Storage:        storageInterface,
+		Versioner:      etcdstorage.APIObjectVersioner{},
+		Type:           objectType,
+		ResourcePrefix: resourcePrefix,
```

Comment:
Would it be possible to remove resourcePrefix from the cacher as well? That doesn't seem like something the caching layer should need to know